### ISSUE TITLE
OAuth Suggestion: SVC and Universal Linking with SPTAuth loginURL

### ISSUE THREAD

**Issue by michaelmerrill** [spotify/ios-sdk#658]
There are a few issues for this beta concerning OAuth right now. May I make a suggestion?

It sounds like Spotify is wanting devs to use SPTAuthViewController to handle login. The problem with this, however, is issue #505 (At least, I believe this is the main issue). On top of that, it looks like SPTAuthViewController is performing a fast app switch under the hood using a universal link. I know this is debatable, but FAS is poor user experience due to the amount of alerts that now inform users between apps. There are also issues with the lack of customizability, #290 , which I know is an enhancement planned for down the road.

The other alternative to FAS would be Safari View Controller which many people believe provides a better user experience for third party login. The benefits to using SVC are described [here](https://developers.facebook.com/blog/post/2015/10/29/Facebook-Login-iOS9/).

The issue with using SVC currently is that `[[SPTAuth defaultInstance] loginURL]` is a universal link before it's a hyperlink. I've built a work around that puts together the http string and does what I need, but all that code could be replaced if we removed the universal link option from `[[SPTAuth defaultInstance] loginURL]` and keep it assigned to the hyperlink.

Then we can use SVC to do OAuth with three lines of code:

`NSURL *loginURL = [[SPTAuth defaultInstance] loginURL];
self.svc = [[SFSafariViewController alloc] initWithURL:loginURL];
[self presentViewController:self.svc animated:true completion:nil];`

This would give us a working OAuth solution while Spotify can work on perfecting SPTAuthViewController.

TL;DR: Prevent `[[SPTAuth defaultInstance] loginURL]` from being a universal link. Or add another property to SPTAuth for getting the hyperlink for oauth.

*** Please let me know if I'm wrong on any of this.

**Comment by ardaerzin** [comment]
I have similar questions in #646

**Comment by idelfonsog2** [comment]
Has this issue been resolved with the latest beta-24?
